gen/calc.py

```diff
@@ -575,6 +575,30 @@ def fz_err(msg):
             validate_int_in_range(port, 1, 65535)
 
 
+def calculate_scarce_resources(enable_scarce_resources):
+    return ','.join(enable_scarce_resources)
+
+
+def calculate_filter_gpu_resources(enable_scarce_resources):
+    if len(enable_scarce_resources) == 0:
+        return 'false'
+    if enable_scarce_resources[0] == 'gpus':
+        return 'true'
+    return 'false'
+
+
+def validate_scarce_resources(enable_scarce_resources):
```

Comment:
Since we have both `scarce_resources` and `enable_scarce_resources` params, let's call this `validate_enable_scarce_resources()` so more obvious about which param it's validating.